habitat/tasks/rearrange/rearrange_grasp_manager.py

```diff
@@ -5,118 +5,232 @@
+        )
 
-    def update(self):
+    def update(self) -> None:
+        """Reset the collision group of the grasped object if its distance to the end effector exceeds a threshold.
+
+        Used to wait for a dropped object to clear the end effector's proximity before re-activating collisions between them.
+        """
         if self._leave_info is not None:
             ee_pos = self._sim.robot.ee_transform.translation
             dist = np.linalg.norm(ee_pos - self._leave_info[0])
             if dist >= self._leave_info[1]:
-                self.snap_rigid_obj.override_collision_group(
-                    CollisionGroups.Default
+                rigid_obj = (
+                    self._sim.get_rigid_object_manager().get_object_by_id(
+                        self._leave_info[2]
+                    )
                 )
+                rigid_obj.override_collision_group(CollisionGroups.Default)
+            self._leave_info = None
 
-    def desnap(self, force=False):
-        """Removes any hold constraints currently active."""
+    def desnap(self, force=False) -> None:
+        """Removes any hold constraints currently active.
+
+        :param force: If True, reset the collision group of the now released object immediately instead of waiting for its distance from the end effector to reach a threshold.
+        """
         if len(self._snap_constraints) == 0:
             # No constraints to unsnap
             self._snapped_obj_id = None
+            self._snapped_marker_id = None
             return
 
         if self._snapped_obj_id is not None:
             obj_bb = get_aabb(self.snap_idx, self._sim)
-            if force:
-                self.snap_rigid_obj.override_collision_group(
-                    CollisionGroups.Default
-                )
-            else:
-                self._leave_info = (
-                    self._sim.get_translation(self._snapped_obj_id),
-                    max(obj_bb.size_x(), obj_bb.size_y(), obj_bb.size_z()),
-                )
+            if obj_bb is not None:
+                if force:
+                    self.snap_rigid_obj.override_collision_group(
+                        CollisionGroups.Default
+                    )
+                else:
+                    self._leave_info = (
+                        self.snap_rigid_obj.translation,
+                        max(obj_bb.size_x(), obj_bb.size_y(), obj_bb.size_z()),
+                        self._snapped_obj_id,
+                    )
 
         for constraint_id in self._snap_constraints:
             self._sim.remove_rigid_constraint(constraint_id)
         self._snap_constraints = []
 
         self._snapped_obj_id = None
+        self._snapped_marker_id = None
 
     @property
-    def snap_idx(self):
+    def snap_idx(self) -> int:
+        """The index of the grasped RigidObject."""
         return self._snapped_obj_id
 
     @property
-    def snap_rigid_obj(self):
+    def snapped_marker_id(self) -> str:
```

Comment:
All marker "ids" are actually strings, so maybe should be refactored to "key"

Reply:
What do you mean by this?